src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_MoveFile.cs

```diff
@@ -0,0 +1,310 @@
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath1.folder}"" FilePath=""{documentPath1.filePath}""> 
+namespace [||]{declaredNamespace}
+{{    
+    class Class1
+    {{
+    }}
+}}  
+        </Document>        
+        <Document Folders=""{documentPath2.folder}"" FilePath=""{documentPath2.filePath}""> 
+namespace Foo
+{{    
+    class Class2
+    {{
+    }}
+}}  
+        </Document>  
+    </Project>
+</Workspace>";
+            await TestMoveFileToMatchNamespace(code, expectedFolders);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task MoveFile_MultipleAction2()
+        {
+            // current path is <root>\
+            // expected new paths are:
+            // 1. <root>\B\C\D\E\
+            // 2. <root>\B.C\D\E\
+            // 3. <root>\B\C.D\E\
+
+            var defaultNamespace = "A";
+            var declaredNamespace = "A.B.C.D.E";
+
+            var expectedFolders = new List<string[]>();
+            expectedFolders.Add(new[] { "B", "C", "D", "E" });
+            expectedFolders.Add(new[] { "B.C", "D", "E" });
+            expectedFolders.Add(new[] { "B", "C.D", "E" });
+
+            var documentPath1 = CreateDocumentFilePath(Array.Empty<string>(), "File1.cs");
+            var documentPath2 = CreateDocumentFilePath(new[] { "B", "C.D" }, "File2.cs");   // file2 is in <root>\B\C.D\
+            var documentPath3 = CreateDocumentFilePath(new[] { "B.C" }, "File3.cs");   // file3 is in <root>\B.C\
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath1.folder}"" FilePath=""{documentPath1.filePath}""> 
+namespace [||]{declaredNamespace}
+{{    
+    class Class1
+    {{
+    }}
+}}  
+        </Document>        
+        <Document Folders=""{documentPath2.folder}"" FilePath=""{documentPath2.filePath}""> 
+namespace Foo
+{{    
+    class Class2
+    {{
+    }}
+}}  
+        </Document>      
+        <Document Folders=""{documentPath3.folder}"" FilePath=""{documentPath3.filePath}""> 
+namespace Foo
+{{    
+    class Class2
+    {{
+    }}
+}}  
+        </Document>  
+    </Project>
+</Workspace>";
+            await TestMoveFileToMatchNamespace(code, expectedFolders);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task MoveFile_FromOneFolderToAnother1()
+        {
+            var defaultNamespace = "A";
+            var declaredNamespace = "A.B.C.D.E";
+
+            var expectedFolders = new List<string[]>();
```

Comment:
if i was involved earily on, I probably would have suggested to just open an issue and do this kind of touch later on :) anyway, 👍